controllers/membership_controller.go

```diff
@@ -0,0 +1,152 @@
+
+	membershipID := c.Param("id")
+
+	membershipObject.InsertID(membershipID, &err)
+	if err.IsError() {
+		response.ErrorOcurred(&err)
+		return c.JSON(response.StatusCode, response)
+	}
+
+	database.GetMembership(&membershipObject, &err)
+	if err.IsError() {
+		response.ErrorOcurred(&err)
+		return c.JSON(response.StatusCode, response)
+	}
+
+	membershipObject.ToReadableMembership(&readableMembership)
+
+	response.Success(http.StatusOK, "Successfully retrieved membership", readableMembership)
+	return c.JSON(response.StatusCode, response)
+}
+
+// CreateMembershipController creates a new membership
+func CreateMembershipController(c echo.Context) error {
+	var response models.GeneralResponse
+	var err models.CustomError
+
+	var readableMembership models.ReadableMembership
+	var membershipObject models.Membership
+
+	err.ErrorMessage = c.Bind(&readableMembership)
+	if err.IsError() {
+		err.StatusCode = http.StatusBadRequest
+		err.ErrorReason = "Invalid request body"
+		response.ErrorOcurred(&err)
+		return c.JSON(response.StatusCode, response)
+	}
+
+	readableMembership.ToMembershipObject(&membershipObject)
+	if err.IsError() {
+		response.ErrorOcurred(&err)
+		return c.JSON(response.StatusCode, response)
+	}
+
+	database.CreateMembership(&membershipObject, &err)
+	if err.IsError() {
+		response.ErrorOcurred(&err)
+		return c.JSON(response.StatusCode, response)
+	}
+
+	membershipObject.ToReadableMembership(&readableMembership)
+
+	response.Success(http.StatusCreated, "Successfully created a new membership", readableMembership)
+	return c.JSON(response.StatusCode, response)
+}
+
+// Update membership
+func UpdateMembershipController(c echo.Context) error {
+	var response models.GeneralResponse
+	var err models.CustomError
+
+	var updatedMembership models.Membership
+
+	membershipID := c.Param("id")
+
+	err.ErrorMessage = c.Bind(&updatedMembership)
+	if err.IsError() {
+		err.StatusCode = http.StatusBadRequest
+		err.ErrorReason = "Invalid request body"
+		response.ErrorOcurred(&err)
+		return c.JSON(response.StatusCode, response)
+	}
+
+	updatedMembership.InsertID(membershipID, &err)
+	if err.IsError() {
+		response.ErrorOcurred(&err)
+		return c.JSON(response.StatusCode, response)
+	}
+
+	database.UpdateMembership(&updatedMembership, &err)
+	if err.IsError() {
+		response.ErrorOcurred(&err)
+		return c.JSON(response.StatusCode, response)
+	}
+
+	response.Success(http.StatusOK, "Successfully updated Membership", updatedMembership)
+	return c.JSON(response.StatusCode, response)
+}
+
+// Delete Membership
+func DeleteMembershipController(c echo.Context) error {
+	var response models.GeneralResponse
+	var err models.CustomError
+
+	membershipID := c.Param("id")
+
+	var deletedMembership models.Membership
+	deletedMembership.InsertID(membershipID, &err)
+	if err.IsError() {
+		response.ErrorOcurred(&err)
+		return c.JSON(response.StatusCode, response)
+	}
+
+	database.DeleteMembership(&deletedMembership, &err)
```

Comment:
data yang ingin dihapus dicari terlebih dahulu. sehingga ketika data yang ingin dihapus tidak ada (not exist) akan mengembalikan response record not found. contohnya dapat dilihat pada user_controller.go